Add introductory text for models in documentation (#2416)
nest/nest-simulator: doc/htmldoc/ref_material/glossary.rst

 E_K
   Potassium reversal potential in Millivolts (mV).

.. _model_terms:

Terms for models in NEST
------------------------

.. glossary::
 :sorted:

 iaf

>    Integrate and fire. Also known in other sources as `IF`.

Reviewer: ```suggestion
   Integrate-and-fire. Also known in other sources as `IF`.
```

 gif
   Generalized integrate and fire. From Gersnter lab.

 glif

>    Generalized leaky integrate and fire. From Allen institute.

Reviewer: ```suggestion
   Generalized leaky integrate-and-fire. From the Allen institute.
```

 cond
   Conductance-based. Also known in other sources as `COBA`.

 psc
   Post-synaptic current (current-based). Also known in other sources as `CUBA`.

 hh
   Hodgkin Huxley.

 aeif
   Adaptive exponential integrate and fire. Also known in other sources as `AdEx`.

 ht
   Hill and Tononi.

 pp
   Point process.

 in
   Inhibitory.

 ex
   Excitatory.

 stdp

>    Spike-timing dependent plasticity synapse.

Reviewer: ```suggestion
   Spike-timing dependent plasticity.
```

Reviewer: To make it consistent with the short-term plasticity entry.

 st

>    Short term plasticity.

Reviewer: ```suggestion
   Short-term plasticity.
```

 psp
   Post-synaptic potential.

 sfa
   Spike-frequency adaptation.

 cm

>    compartmental model.

Reviewer: ```suggestion
   Compartmental model.
```

Reviewer: For consistency.

Other abbreviations
-------------------

.. glossary::
 :sorted:

 rng
   Random number generator.

 wfr
   Waveform relaxation method.

 MAM
   Multi-area model.

 mpi
   Message passing interface.

 vp
   Virtual process.

> Commonly used terms in NEST

Author: Could you please either create an issue regarding the glossary errors, or make a PR with suggested fixes?

> Commonly used terms in NEST

Reviewer: Ok

 soma
   Cell body of the neuron.

>  absolute refractory

Reviewer: ```suggestion
 absolute refractory period
```

>    An interval after a neuron fires a spike to prevent it from firing a spike again.

Reviewer: ```suggestion
   Interval directly following a spike emission in which the sender neuron cannot fire again.
```

Reviewer: The reason for the refractory period is of biological nature (depletion of neurotransmitters etc.). As the description was formulated it sounds as if there was a technical reason we would need to prevent follow-up spikes.
